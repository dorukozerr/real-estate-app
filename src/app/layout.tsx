import type { Metadata } from "next";
import { Fira_Sans_Condensed } from "next/font/google";
import Link from "next/link";
import { ThemeProvider } from "@/components/theme/theme-provider";
import { ThemeToggler } from "@/components/theme/theme-toggler";
import { Toaster } from "@/components/ui/sonner";

import "swiper/css";
import "swiper/css/effect-cards";
import "swiper/css/navigation";
import "swiper/css/pagination";
import "./globals.css";

const firaSans = Fira_Sans_Condensed({
  variable: "--font-fira",
  weight: ["100", "200", "300", "400", "500", "600", "700", "800", "900"],
  subsets: ["latin-ext"],
});

export const metadata: Metadata = {
  title: "Emluck",
  description: "Basit bir emlakçı sitesi",
};

const RootLayout = ({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) => (
  <html lang="en" suppressHydrationWarning={true} className="h-full">
    <body
      suppressHydrationWarning={true}
      className={`${firaSans.variable} h-full`}
    >
      <ThemeProvider
        attribute="class"
        defaultTheme="system"
        enableSystem
        disableTransitionOnChange
      >
        <div className="relative flex h-full w-full flex-col overflow-hidden">
          <header className="relative z-10 mx-auto flex h-[70px] min-h-[70px] w-screen max-w-[1440px] items-center justify-between border-b border-border px-4">
            <Link href="/">
              <h2 className="text-2xl font-bold sm:text-4xl">Emluck</h2>
            </Link>
            <ThemeToggler />
          </header>
          <main className="relative z-10 flex w-full flex-1 flex-col overflow-hidden">
            {children}
          </main>
          <div className="absolute -z-10 h-full w-full animate-[spin_20000ms_linear_infinite]">
            <div className="absolute left-1/2 top-1/2 h-[100vw] w-[100vh] -translate-x-1/2 -translate-y-1/2 rounded-full bg-gradient-to-r from-pink-200 to-green-200 blur-3xl dark:from-[#3B4E68] dark:to-[#5F5255] sm:h-[40vh] sm:w-[100vw]"></div>
          </div>
        </div>
        <Toaster />
      </ThemeProvider>
    </body>
  </html>
);

export default RootLayout;
