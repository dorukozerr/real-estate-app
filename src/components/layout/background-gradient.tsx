export const BackgroundGradient = () => {
  return (
    <svg
      xmlnsXlink="http://www.w3.org/1999/xlink"
      width="2162"
      height="1912"
      viewBox="0 0 2162 1912"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g opacity="0.4" fill="none">
        <g filter="url(#filter0_f_117_3888)" fill="none">
          <path
            d="M1065.81 712.997C1157.29 712.997 1179.87 858.777 1179.87 950.078C1179.87 1041.38 1397.93 1096.77 1271.5 1155.44C1212.82 1182.67 1167.39 1175.4 1105.86 1155.44C959.701 1108.03 900.178 969.615 900.178 878.313C900.178 787.012 974.336 712.997 1065.81 712.997Z"
            fill="#FF90EA"
          ></path>
        </g>
        <g filter="url(#filter1_f_117_3888)" fill="none">
          <path
            d="M1086.35 1175.62C1136.66 1233.92 1179.22 1327.83 1177.97 1404.83C1173.99 1650.03 629.281 1426.28 512.827 1210.47C413.721 1026.81 314.541 660.88 522.668 645.505C654.653 635.755 797.866 678.67 854.595 798.24C888.264 869.207 874.84 986.005 922.001 1048.82C958.893 1097.96 1046.2 1129.1 1086.35 1175.62Z"
            fill="#FFDEDA"
          ></path>
        </g>
        <g opacity="0.7" filter="url(#filter2_f_117_3888)" fill="none">
          <circle
            cx="314.934"
            cy="314.934"
            r="314.934"
            transform="matrix(1 0 0 -1 1132.13 1210.87)"
            fill='url("#paint0_linear_117_3888")'
          ></circle>
        </g>
        <g filter="url(#filter3_f_117_3888)" fill="none">
          <circle
            cx="76.8912"
            cy="76.8912"
            r="76.8912"
            transform="matrix(1 0 0 -1 1198.13 1014.15)"
            fill="#0094FF"
          ></circle>
        </g>
        <g filter="url(#filter4_f_117_3888)" fill="none">
          <path
            d="M1065.81 326.997C1157.29 326.997 1179.87 472.777 1179.87 564.078C1179.87 655.38 1397.93 710.767 1271.5 769.442C1212.82 796.672 1167.39 789.4 1105.86 769.442C959.701 722.032 900.178 583.615 900.178 492.313C900.178 401.012 974.336 326.997 1065.81 326.997Z"
            fill="#FF90EA"
          ></path>
        </g>
        <g filter="url(#filter5_f_117_3888)" fill="none">
          <path
            d="M1086.35 789.619C1136.66 847.922 1179.22 941.833 1177.97 1018.83C1173.99 1264.03 629.281 1040.28 512.827 824.471C413.721 640.81 314.541 274.88 522.668 259.505C654.653 249.755 797.866 292.67 854.595 412.24C888.264 483.207 874.84 600.005 922.001 662.82C958.893 711.957 1046.2 743.1 1086.35 789.619Z"
            fill="#FFDEDA"
          ></path>
        </g>
        <g filter="url(#filter6_f_117_3888)" fill="none">
          <circle
            cx="314.934"
            cy="314.934"
            r="314.934"
            transform="matrix(1 0 0 -1 1132.13 824.867)"
            fill='url("#paint1_linear_117_3888")'
          ></circle>
        </g>
        <g filter="url(#filter7_f_117_3888)" fill="none">
          <circle
            cx="76.8912"
            cy="76.8912"
            r="76.8912"
            transform="matrix(1 0 0 -1 1198.13 628.154)"
            fill="#0094FF"
          ></circle>
        </g>
      </g>
      <defs>
        <filter
          id="filter0_f_117_3888"
          x="500.178"
          y="312.997"
          width="1210.75"
          height="1260.24"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="200"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter1_f_117_3888"
          x="0.206543"
          y="244.17"
          width="1577.79"
          height="1667.71"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="200"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter2_f_117_3888"
          x="732.133"
          y="181"
          width="1429.87"
          height="1429.87"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="200"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter3_f_117_3888"
          x="898.131"
          y="560.371"
          width="753.782"
          height="753.782"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="150"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter4_f_117_3888"
          x="705.178"
          y="131.997"
          width="800.748"
          height="850.244"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="97.5"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter5_f_117_3888"
          x="205.207"
          y="63.1699"
          width="1167.79"
          height="1257.71"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="97.5"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter6_f_117_3888"
          x="937.133"
          y="0.000244141"
          width="1019.87"
          height="1019.87"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="97.5"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <filter
          id="filter7_f_117_3888"
          x="1003.13"
          y="279.371"
          width="543.782"
          height="543.782"
          filterUnits="userSpaceOnUse"
          colorInterpolationFilters="sRGB"
        >
          <feFlood floodOpacity="0" result="BackgroundImageFix"></feFlood>
          <feBlend
            mode="normal"
            in="SourceGraphic"
            in2="BackgroundImageFix"
            result="shape"
          ></feBlend>
          <feGaussianBlur
            stdDeviation="97.5"
            result="effect1_foregroundBlur_117_3888"
          ></feGaussianBlur>
        </filter>
        <linearGradient
          id="paint0_linear_117_3888"
          x1="314.934"
          y1="0"
          x2="314.934"
          y2="629.867"
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#8BB5FF"></stop>
          <stop offset="1" stopColor="#E1FBFF"></stop>
        </linearGradient>
        <linearGradient
          id="paint1_linear_117_3888"
          x1="314.934"
          y1="0"
          x2="314.934"
          y2="629.867"
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#8BB5FF"></stop>
          <stop offset="1" stopColor="#E1FBFF"></stop>
        </linearGradient>
        <linearGradient
          id="paint0_linear_117_3888"
          x1="314.934"
          y1="0"
          x2="314.934"
          y2="629.867"
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#8BB5FF"></stop>
          <stop offset="1" stopColor="#E1FBFF"></stop>
        </linearGradient>
        <linearGradient
          id="paint1_linear_117_3888"
          x1="314.934"
          y1="0"
          x2="314.934"
          y2="629.867"
          gradientUnits="userSpaceOnUse"
        >
          <stop stopColor="#8BB5FF"></stop>
          <stop offset="1" stopColor="#E1FBFF"></stop>
        </linearGradient>
      </defs>
    </svg>
  );
};
