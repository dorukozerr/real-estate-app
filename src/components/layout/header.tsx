"use client";

import Link from "next/link";
import { useSearchParams } from "next/navigation";
import { useTheme } from "next-themes";
import { HamburgerMenuIcon } from "@radix-ui/react-icons";
import { ThemeToggler } from "@/components/theme/theme-toggler";
import { Button } from "@/components/ui/button";
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuGroup,
  DropdownMenuItem,
  DropdownMenuLabel,
  DropdownMenuPortal,
  DropdownMenuSeparator,
  DropdownMenuSub,
  DropdownMenuSubContent,
  DropdownMenuSubTrigger,
  DropdownMenuTrigger,
} from "@/components/ui/dropdown-menu";

export const Header = () => {
  const searchParams = useSearchParams();
  const { setTheme } = useTheme();

  const mode = searchParams.get("mode") || "";

  const themeOptions = [
    { label: "Aydınlık", onClick: () => setTheme("light") },
    { label: "Karanlık", onClick: () => setTheme("dark") },
    { label: "Sistem", onClick: () => setTheme("system") },
  ];

  const links = [
    { href: "/", label: "Anasayfa" },
    { href: "/about-us", label: "Biz Kimiz" },
    {
      href: "/listing?mode=sale&page=1&size=9",
      label: "Satılık İlanlar",
      key: "sale",
    },
    {
      href: "/listing?mode=rent&page=1&size=9",
      label: "Kiralık İlanlar",
      key: "rent",
    },
  ];

  return (
    <header className="relative z-10 mx-auto flex h-[70px] min-h-[70px] w-full max-w-[1440px] items-center justify-between border-b border-border px-4">
      <Link href="/">
        <h2 className="text-2xl font-bold sm:text-4xl">Emluck</h2>
      </Link>
      <div className="hidden items-center justify-start gap-4 sm:flex">
        {links
          .filter(({ label }) => label !== "Anasayfa")
          .map(({ href, label, key }, linkIndex) => (
            <Link href={href} key={`headerLink-${linkIndex}`}>
              <Button variant={key === mode ? "outline" : "ghost"}>
                {label}
              </Button>
            </Link>
          ))}
        <ThemeToggler />
      </div>
      <div className="block sm:hidden">
        <DropdownMenu>
          <DropdownMenuTrigger asChild>
            <Button variant="outline" size="icon">
              <HamburgerMenuIcon className="h-[1.2rem] w-[1.2rem]" />
            </Button>
          </DropdownMenuTrigger>
          <DropdownMenuContent className="w-56">
            <DropdownMenuLabel>Menü</DropdownMenuLabel>
            <DropdownMenuSeparator />
            <DropdownMenuGroup>
              {links.map(({ href, label, key }, linkIndex) => (
                <Link href={href} key={`mobileMenuItem-${linkIndex}`}>
                  <DropdownMenuItem>
                    <span className={key === mode ? "underline" : ""}>
                      {label}
                    </span>
                  </DropdownMenuItem>
                </Link>
              ))}
            </DropdownMenuGroup>
            <DropdownMenuSeparator />
            <DropdownMenuGroup>
              <DropdownMenuSub>
                <DropdownMenuSubTrigger>
                  <span>Tema</span>
                </DropdownMenuSubTrigger>
                <DropdownMenuPortal>
                  <DropdownMenuSubContent>
                    {themeOptions.map(
                      ({ label, onClick }, themeOptionIndex) => (
                        <DropdownMenuItem
                          onClick={onClick}
                          key={`mobileThemeToggler-${themeOptionIndex}`}
                        >
                          <span>{label}</span>
                        </DropdownMenuItem>
                      )
                    )}
                  </DropdownMenuSubContent>
                </DropdownMenuPortal>
              </DropdownMenuSub>
            </DropdownMenuGroup>
          </DropdownMenuContent>
        </DropdownMenu>
      </div>
    </header>
  );
};
