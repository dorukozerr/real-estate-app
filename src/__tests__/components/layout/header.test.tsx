import { ThemeProvider } from "next-themes";
import { render } from "@testing-library/react";
import { Header } from "@/components/layout/header";

describe("Header", () => {
  const originalInnerWidth = window.innerWidth;

  const setViewportWidth = (width: number) => {
    Object.defineProperty(window, "innerWidth", {
      writable: true,
      configurable: true,
      value: width,
    });

    window.getComputedStyle = (element: Element) => {
      const isMobile = width < 640;
      const isMobileNav = element.getAttribute("data-testid") === "mobile-nav";
      const isDesktopNav =
        element.getAttribute("data-testid") === "desktop-nav";

      return {
        display: (() => {
          if (isMobile) {
            return isMobileNav ? "block" : "none";
          } else {
            return isDesktopNav ? "flex" : "none";
          }
        })(),
        visibility: "visible",
      } as CSSStyleDeclaration;
    };

    window.dispatchEvent(new Event("resize"));
  };

  const renderHeader = () =>
    render(
      <ThemeProvider>
        <Header />
      </ThemeProvider>
    );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  afterEach(() => {
    Object.defineProperty(window, "innerWidth", {
      writable: true,
      configurable: true,
      value: originalInnerWidth,
    });
  });

  it("renders correctly", () => {
    const { container } = renderHeader();

    expect(container).toMatchSnapshot();
  });

  it("desktop navbar is hidden if screen is sm and mobile navbar is visible", () => {
    setViewportWidth(500);

    const { getByTestId } = renderHeader();

    const mobileNav = getByTestId("mobile-nav");
    const desktopNav = getByTestId("desktop-nav");

    expect(window.getComputedStyle(mobileNav).display).toBe("block");
    expect(window.getComputedStyle(desktopNav).display).toBe("none");
  });

  it("mobile navbar is hidden if screen is bigger than sm and desktop navbar is visible", () => {
    setViewportWidth(1000);

    const { getByTestId } = renderHeader();

    const mobileNav = getByTestId("mobile-nav");
    const desktopNav = getByTestId("desktop-nav");

    expect(window.getComputedStyle(mobileNav).display).toBe("none");
    expect(window.getComputedStyle(desktopNav).display).toBe("flex");
  });
});
