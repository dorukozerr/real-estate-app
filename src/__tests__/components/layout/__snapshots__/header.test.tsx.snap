// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <header
    class="relative z-10 mx-auto flex h-[70px] min-h-[70px] w-full max-w-[1440px] items-center justify-between border-b border-border px-4"
  >
    <a
      href="/"
    >
      <h2
        class="text-2xl font-bold sm:text-4xl"
      >
        Emluck
      </h2>
    </a>
    <div
      class="hidden items-center justify-start gap-4 sm:flex"
      data-testid="desktop-nav"
    >
      <a
        href="/about-us"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
        >
          Biz Kimiz
        </button>
      </a>
      <a
        href="/listing?mode=sale&page=1&size=9"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
        >
          Satılık İlanlar
        </button>
      </a>
      <a
        href="/listing?mode=rent&page=1&size=9"
      >
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2"
        >
          Kiralık İlanlar
        </button>
      </a>
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
        data-state="closed"
        id="radix-:r0:"
        type="button"
      >
        <svg
          class="lucide lucide-sun h-[1.2rem] w-[1.2rem] rotate-0 scale-100 transition-all dark:-rotate-90 dark:scale-0"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="4"
          />
          <path
            d="M12 2v2"
          />
          <path
            d="M12 20v2"
          />
          <path
            d="m4.93 4.93 1.41 1.41"
          />
          <path
            d="m17.66 17.66 1.41 1.41"
          />
          <path
            d="M2 12h2"
          />
          <path
            d="M20 12h2"
          />
          <path
            d="m6.34 17.66-1.41 1.41"
          />
          <path
            d="m19.07 4.93-1.41 1.41"
          />
        </svg>
        <svg
          class="lucide lucide-moon absolute h-[1.2rem] w-[1.2rem] rotate-90 scale-0 transition-all dark:rotate-0 dark:scale-100"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
          />
        </svg>
        <span
          class="sr-only"
        >
          Toggle theme
        </span>
      </button>
    </div>
    <div
      class="block sm:hidden"
      data-testid="mobile-nav"
    >
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-10 w-10"
        data-state="closed"
        id="radix-:r2:"
        type="button"
      >
        <svg
          class="h-[1.2rem] w-[1.2rem]"
          fill="none"
          height="15"
          viewBox="0 0 15 15"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M1.5 3C1.22386 3 1 3.22386 1 3.5C1 3.77614 1.22386 4 1.5 4H13.5C13.7761 4 14 3.77614 14 3.5C14 3.22386 13.7761 3 13.5 3H1.5ZM1 7.5C1 7.22386 1.22386 7 1.5 7H13.5C13.7761 7 14 7.22386 14 7.5C14 7.77614 13.7761 8 13.5 8H1.5C1.22386 8 1 7.77614 1 7.5ZM1 11.5C1 11.2239 1.22386 11 1.5 11H13.5C13.7761 11 14 11.2239 14 11.5C14 11.7761 13.7761 12 13.5 12H1.5C1.22386 12 1 11.7761 1 11.5Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </header>
</div>
`;
